src/Generator/EnumGenerator/Name.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator;
+
+use function strpos;
+use function strrpos;
+use function substr;
+
+final class Name
+{
+    private string $name;
+
+    private ?string $namespace;
+
+    private function __construct(string $name, ?string $namespace)
+    {
+        $this->name      = $name;
+        $this->namespace = $namespace;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getNamespace(): ?string
+    {
+        return $this->namespace;
+    }
+
+    public static function fromName(string $name): self
+    {
+        $namespace = null;
+        if (false !== strpos($name, '\\')) {
+            $namespace = substr($name, 0, strrpos($name, '\\'));
+            $name      = substr($name, strrpos($name, '\\') + 1);
+        }
```

Comment:
```suggestion
        $nsPosition = strrpos($name, '\\');
        if (false !== $nsPosition) {
            $namespace = substr($name, 0, $nsPosition);
            $name      = substr($name, $nsPosition + 1);
        }
```

This will correct the error, and prevent duplicate operations, making it perform faster.